File: tools/python-execute-eval/results/grok-4.7/csv-wrangle.py
import csv
import io


def main(params):
    by_sku = {}
    for row in csv.DictReader(io.StringIO(params["csv"])):
        sku = row["sku"]
        qty = int(row["qty"])
        price = float(row["price"])
        bucket = by_sku.setdefault(sku, {"qty": 0, "revenue": 0.0})
        bucket["qty"] += qty
        bucket["revenue"] += qty * price
    return {"by_sku": by_sku}

File: tools/python-execute-eval/results/grok-4.7/csv-wrangle.mjs
export default function main(params) {
	const bySku = {}
	const [header, ...lines] = params.csv.trim().split('\n')
	const columns = header.split(',')
	const skuIndex = columns.indexOf('sku')
	const qtyIndex = columns.indexOf('qty')
	const priceIndex = columns.indexOf('price')
	for (const line of lines) {
		if (!line) continue
		const cells = line.split(',')
		const sku = cells[skuIndex]
		const qty = Number(cells[qtyIndex])
		const price = Number(cells[priceIndex])
		const bucket = (bySku[sku] ??= { qty: 0, revenue: 0 })
		bucket.qty += qty
		bucket.revenue += qty * price
	}
	return { by_sku: bySku }
}
